### PR TITLE
Add action for generating docs for pages (Close #5)

### LINKED ISSUE #5: Add Github Action to generate doc site

### PR DESCRIPTION
**Pull request by rlh1994**
## Description & motivation
Adds a GH action to generate the docs based on the snowflake connection, then push these to the `gh_pages` branch (we will need someone to swap the pages settings to look at this branch instead of main before we can really use this). Should save us having to do it in the future and mean all releases now have the same schema and naming regardless of who releases it.

## Checklist
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have raised a [documentation](https://github.com/snowplow/documentation) PR if applicable (N/A)

### CODE REVIEW

File: .github/workflows/publish-gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
+name: generate-docs-pages
+
+on:
+  push:
+    branches:
+      - main
```
Comment (emielver) [inline .github/workflows/publish-gh-pages.yml:6]: Do we want to do this on main or on a release branch? What if this fails for whatever reason?

File: .github/workflows/publish-gh-pages.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+concurrency: dbt_integration_tests
+
+env:
+  # Set profiles.yml directory
+  DBT_PROFILES_DIR: ./ci
+
+  # Redshift Connection
```
Comment (emielver) [inline .github/workflows/publish-gh-pages.yml:14]: Why not remove the credentials/steps that are unused instead of commenting out?

**Review by emielver — COMMENTED** [review]
Some open questions but otherwise all good!

File: .github/workflows/publish-gh-pages.yml (hunk 2 above)
Comment (rlh1994) [reply .github/workflows/publish-gh-pages.yml:14]: Keeping for incase we ever need to change what we build from

File: .github/workflows/publish-gh-pages.yml (hunk 1 above)
Comment (rlh1994) [reply .github/workflows/publish-gh-pages.yml:6]: If it fails can run manually and gh_pages branch is unprotected worst case

**Review by emielver — APPROVED** [review]
LGTM!

**Pull request merged**